File: algorithms/python/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        value_index = {}
        for index, value in enumerate(nums):
            diff = target - value
            if diff in value_index:
                return [min(value_index[diff], index), max(value_index[diff], index)]
            value_index[value] = index
        return []

        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

File: algorithms/go/1. Two Sum.go
package main

func twoSum(nums []int, target int) []int {
	memoIndex := make(map[int]int)
	for i, v := range nums {
		remain := target - v
		if _, exists := memoIndex[remain]; exists {
			return []int{memoIndex[remain], i}
		}
		memoIndex[v] = i
	}
	return nil
}
